File: 2714-left-and-right-sum-differences/left-and-right-sum-differences.scala
object Solution {
    def leftRightDifference(nums: Array[Int]): Array[Int] = {
        def iter(nums: Array[Int], leftSum: Int, rightSum: Int, result: Array[Int]): Array[Int] = 
            nums.headOption match
                case Some(num) =>
                    val updatedLeftSum = leftSum + num
                    val updatedRightSum = rightSum - num
                    val resultValue = (updatedLeftSum - rightSum).abs
                    iter(nums.tail, updatedLeftSum, updatedRightSum, result :+ resultValue)
                case None      => result
        iter(nums, 0, nums.sum, Array.empty[Int])
    }
}

File: 2714-left-and-right-sum-differences/left-and-right-sum-differences.java
public class Solution {
    public int[] leftRightDifference(int[] nums) {
        int n = nums.length;
        int[] differenceArray = new int[n];
        int sumLeft = 0;
        int sumRight = 0;

        for(int i = 0; i < n; i++){
            sumRight += nums[i];
        }

        for(int i = 0; i < n; i++){
            sumLeft += nums[i];
            differenceArray[i] = Math.abs(sumLeft - sumRight);
            sumRight -= nums[i];
        }

        return differenceArray;
    }
}